client.go

```diff
@@ -1,39 +1,116 @@
 package kinesis
 
 import (
+	"encoding/json"
+	"io/ioutil"
 	"net/http"
 	"os"
+	"strings"
+	"time"
 )
 
 const (
 	ACCESS_ENV_KEY  = "AWS_ACCESS_KEY"
 	SECRET_ENV_KEY  = "AWS_SECRET_KEY"
 	REGION_ENV_NAME = "AWS_REGION_NAME"
+
+	AWS_METADATA_SERVER = "169.254.169.254"
+	AWS_IAM_CREDS_PATH  = "/latest/meta-data/iam/security-credentials"
+	AWS_IAM_CREDS_URL   = "http://" + AWS_METADATA_SERVER + AWS_IAM_CREDS_PATH
+
+	AWS_SECURITY_TOKEN_HEADER = "X-Amz-Security-Token"
 )
 
 // Auth store information about AWS Credentials
 type Auth struct {
+	// AccessKey, SecretKey are the standard AWS auth credentials
 	AccessKey, SecretKey, Token string
+	// Expiry indicates the time at which these credentials expire. If this is set
+	// to anything other than the zero value, indicates that the credentials are
+	// temporary (and probably fetched from an IAM role)
+	Expiry time.Time
 }
 
 // Client is like http.Client, but signs all requests using Auth.
 type Client struct {
+	// Auth holds the credentials for this client instance
 	Auth *Auth
 	// The http client to make requests with. If nil, http.DefaultClient is used.
 	Client *http.Client
 }
 
-// New creates a new Client.
+// InferCredentialsFromEnv retrieves auth credentials from environment vars
+func (auth *Auth) InferCredentialsFromEnv() {
+	auth.AccessKey = os.Getenv(ACCESS_ENV_KEY)
+	auth.SecretKey = os.Getenv(SECRET_ENV_KEY)
+}
+
+// InferCredentialsFromMetadata retrieves auth credentials from the metadata
+// server. If an IAM role is associated with the instance we are running on, the
+// metadata server will expose credentials for that role under a known endpoint.
+//
+// TODO: specify custom network (connect, read) timeouts, else this will block
```

Comment:
Should create a issue in repo and remove this comment? Or get the TODO into this PR/